WI #678: Adding use case to work item template (#677)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: supporting_content/usecases/howtocraftusecases.md

    * [Report commitment discount purchases](https://focus.finops.org/use-cases/#modal-column-14638)
    * [Determine Effective Savings Rate](https://focus.finops.org/use-cases/#modal-column-14600)

2. Add the `Use Case Title(s)` to the “Use Case(s)” section in the Work Item

3. Document the use case(s) in the [FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866)
    * Find the next empty row on the `Use Cases` tab of the [FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866)
    * Update Status to “Not Started” and enter your name in “Owner”
    * Review [Requirements](#requirements) and fill out the columns:
    * Supported Feature: Identify and enter the Supported Feature
    * Title: Enter the Use Case title written in step 1
    * Context: Enter additional supporting detail further explaining the use case
    * FOCUS Columns: Enter the columns to be used in the use case SQL query.  This does not have to be perfect, enter your best guess.

>     * OPTIONAL: SQL: SQL query for the use case (this will be reviewed and generated by the FinOps Foundation)
>     * Persona: Enter the relevant Persona(s)
>     * Capability: Enter the relevant Capability(s)

Reviewer: ```suggestion
    * Persona: Enter the relevant Persona(s)
    * Capability: Enter the relevant Capability(s)
    * SQL (optional): SQL query for the use case (this will be reviewed and generated by the FinOps Foundation)
```

# How to Craft Use Cases

## Purpose
This document will help FOCUS contributors create use cases that will be posted on the [Use Case Library](https://focus.finops.org/use-cases/).

This document is for use in generating use cases for 1.2 work items and will be converted into a markdown document and added into the specification.

## Artifacts
[FOCUS Use Case Library website](https://focus.finops.org/use-cases/) - Public website destination for use cases.
[FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866) - All use cases correlated here from work items.

## Requirements

> Reference [FOCUS Use Case Library](https://focus.finops.org/use-cases/) for examples.  All use cases will end up here on the public website.
> 
> Use cases must be built with the [FinOps Persona](https://www.finops.org/framework/personas/) in mind: You will need to select at least 1.
> 
> Use cases must align with a [Capability](https://www.finops.org/framework/personas/) in mind:  You will need to select at least 1.
> 
> Use cases should be brief and clearly stated but do not need to be in the form of a user story or a complete sentence.  The Context section for a use case is where to elaborate and describe it further.
> 
> Use cases must reference [FOCUS columns](https://focus.finops.org/focus-columns/).  Use existing columns and new, upcoming ones.  Add only valuable/pertinent columns to the use case.
> 
> A use case can be accompanied by a SQL query.  This will be built from the columns and we will help you build these.  The Foundation will determine the final version of the SQL query.

Reviewer: Move to bullets.

```suggestion
* Reference [FOCUS Use Case Library](https://focus.finops.org/use-cases/) for examples.  All use cases will end up here on the public website.
* Use cases must be built with the [FinOps Persona](https://www.finops.org/framework/personas/) in mind: You will need to select at least 1.
* Use cases must align with a [Capability](https://www.finops.org/framework/personas/) in mind:  You will need to select at least 1.
* Use cases should be brief and clearly stated but do not need to be in the form of a user story or a complete sentence.  The Context section for a use case is where to elaborate and describe it further.
* Use cases must reference [FOCUS columns](https://focus.finops.org/focus-columns/).  Use existing columns and new, upcoming ones.  Add only valuable/pertinent columns to the use case.
* A use case can be accompanied by a SQL query.  This will be built from the columns and we will help you build these.  The Foundation will determine the final version of the SQL query.
```

## Steps

1. Start with writing the `Use Case Title(s)` for the Work Item.  Use these examples and others from the [FOCUS Use Case Library](https://focus.finops.org/use-cases/) as inspiration:
a. [Analyze the different metered costs for a particular SKU](https://focus.finops.org/use-cases/#modal-column-17972)
b. [Report costs by service category](https://focus.finops.org/use-cases/#modal-column-14650)
c. [Report commitment discount purchases](https://focus.finops.org/use-cases/#modal-column-14638)
d. [Determine Effective Savings Rate](https://focus.finops.org/use-cases/#modal-column-14600)

2. Add the `Use Case Title(s)` to the “Use Case(s)” section in the Work Item

> 3. Document the use case(s) in the [FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866)
> a. Find the next empty row on the `Use Cases` tab of the [FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866)
> b. Update Status to “Not Started” and enter your name in “Owner”
> e. Review [Requirements](#requirements) and fill out the columns:
> f. Supported Feature: Identify and enter the Supported Feature
> g. Title: Enter the Use Case title written in step 1
> h. Context: Enter additional supporting detail further explaining the use case
> i. FOCUS Columns: Enter the columns to be used in the use case SQL query.  This does not have to be perfect, enter your best guess.
> j. OPTIONAL: SQL: SQL query for the use case (this will be reviewed and generated by the FinOps Foundation)
> k. Persona: Enter the relevant Persona(s)
> l. Capability: Enter the relevant Capability(s)

Reviewer: Unfortunately GitHub markdown doesn't support lettered lists; see [here](https://stackoverflow.com/questions/13366820/how-do-you-make-lettered-lists-using-markdown).  I'd suggest using bullets and then putting the letters in the bullet itself.  Also, you need to reletter the list.  So:

```suggestion
3. Document the use case(s) in the [FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866)
   * a. Find the next empty row on the `Use Cases` tab of the [FOCUS Use Case Library document](https://docs.google.com/spreadsheets/d/1y5zeLZXLaEwK9Y9MT3khAHOy2c2hpnSrx4YCutZCQ0A/edit?gid=1245179866#gid=1245179866)
   * b. Update Status to “Not Started” and enter your name in “Owner”
   * c. Review [Requirements](#requirements) and fill out the columns:
   * d. Supported Feature: Identify and enter the Supported Feature
   * e. Title: Enter the Use Case title written in step 1
   * f. Context: Enter additional supporting detail further explaining the use case
   * g. FOCUS Columns: Enter the columns to be used in the use case SQL query.  This does not have to be perfect, enter your best guess
   * h. OPTIONAL: SQL: SQL query for the use case (this will be reviewed and generated by the FinOps Foundation)
   * i. Persona: Enter the relevant Persona(s)
   * j. Capability: Enter the relevant Capability(s)
```
